File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();

        int st = 0 , end = n - 1;

        while(st < end){
            int sum = nums[st] + nums[end];

            if(sum == target){
                return {st+1 , end+1};
                st++ , end--;
            }
            else if(sum > target){
                end--;
            }
            else{
                st++;
            }
        }
        return {};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i , j = 0 , len(nums)-1
        l = []

        while i <= j:
            sum = nums[i] + nums[j]
            if sum == target:
                l.append(i+1)
                l.append(j+1)
                return l
            elif sum > target:
                j -= 1
            else:
                i += 1

        return -1